### PR TITLE
Added support for SubroutineCall during template instantiation

### PR DESCRIPTION
**Pull request by ansharlubis**
Solved the bug mentioned in #1385.

### CODE REVIEW

**Comment by certik** [comment]
The WASM test fails due to https://github.com/lfortran/lfortran/issues/1386, so just uncomment it for now.

File: integration_tests/CMakeLists.txt [hunk 1]
```diff
@@ -675,6 +675,7 @@ RUN(NAME template_array_01b LABELS llvm wasm)
 RUN(NAME template_array_02 LABELS llvm)
 RUN(NAME template_array_02b LABELS llvm wasm)
 RUN(NAME template_array_03 LABELS llvm wasm)
+RUN(NAME template_array_04 LABELS llvm wasm)
```
Comment (certik) [inline integration_tests/CMakeLists.txt:678]:
```suggestion
RUN(NAME template_array_04 LABELS llvm)
```

**Review by czgdp1807 — APPROVED** [review]
LGTM.

**Review by certik — APPROVED** [review]

**Comment by certik** [comment]
Thanks!

**Pull request merged**